accepted/future-releases/constructor-tearoffs/feature-specification.md

```diff
@@ -517,22 +517,25 @@ The grammar changes necessary for these changes are provided separately (as [cha
+If moreover `e` is a constant expression whose static type is a function type `F` or `e` is a type literal, and <code>T<sub>1</sub>..T<sub>k</sub></code> is a list of constant type expressions, then <code>e\<T<sub>1</sub>..T<sub>k</sub>></code> is a constant expression.
 
-*It follows that `F` is a generic function type taking `k` type arguments, and `T1..Tk` satisfy the bounds of `F`, and similarly for the type literal, because otherwise the program would have a compile-time error.*
+*It follows that `F` is a generic function type taking `k` type arguments, and <code>T<sub>1</sub>..T<sub>k</sub></code> satisfy the bounds of `F`, and similarly for the type literal, because otherwise the program would have a compile-time error.*
 
-The following cases are specified elsewhere in this document:
+Assume that <code>*C*.*name*</code> denotes a constructor, and <code>T<sub>1</sub>..T<sub>k</sub></code> is derived from <code>\<typeArguments></code>, for some <code>k >= 0</code>.
+<code>*C*\<T<sub>1</sub>..T<sub>k</sub>>.*name*</code> is then a potentially constant expression if <code>T<sub>j</sub></code> is a potentially constant type expression for each <code>j</code>.
+It is further a constant expression if <code>T<sub>j</sub></code> is a constant type expression for each <code>j</code>.
 
-*Section 'Named constructor tearoffs':
-This section says that expressions of the form <code>*C*\<*typeArgs*>.*name*</code> can be potentially constant and constant expressions.
-Also, the constantness of named constructor tearoffs follows the constantness of the tearoffs of the corresponding constructor functions.*
+_In particular, <code>*C*.*name*</code>, which is the case where <code>k == 0</code>, is always constant when <code>*C*</code> is a non-generic class, and it may or may not be constant if it is generic and the type arguments are inferred._
 
-*Section 'Tearing off constructors from type aliases':
-This section contains several rules about constant expressions: About non-generic type aliases; about generic type aliases that are applied to some actual type arguments; about generic type aliases that are 'proper renames' and that do not receive any actual type arguments; and about generic type aliases that are not 'proper renames' and do not receive any actual type arguments.
-In general, their constantness follows the constantness of specific corresponding constructor functions.*
+If `T` denotes a type variable then `T` is a potentially constant type expression, and a potentially constant expression.
+
+_This is just a simpler way to write something which is already supported: If we wish to use a type variable as an expression in a constant constructor initializer list, we can define <code>typedef F\<X> = X;</code> and express the same thing as <code>F\<T></code>._
```

Comment:
(And that's the only reason to do it. Damn, I wish we could just remove the `Type` type, or move it into `dart:mirrors`.)

Reply:
(We will always have a plethora of ways to specify any particular thing, and writing `F<T>` in a location where we want to write `T` is just one of them.)